### PR TITLE
Fix QSet deprecation warnings

### PR DESCRIPTION
**Pull request by Waqar144**

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -3298,18 +3300,19 @@ bool MainWindow::buildNotesIndex(int noteSubFolderId, bool forceRebuild) {
...
         // check for removed note subfolders
-//#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
+        QList<int> removedNoteSubFolderIdList =
+            QSet<int>(_buildNotesIndexBeforeNoteSubFolderIdList.begin(),
+                      _buildNotesIndexBeforeNoteSubFolderIdList.end())
+                .subtract(QSet<int>(_buildNotesIndexAfterNoteSubFolderIdList.begin(),
+                          _buildNotesIndexAfterNoteSubFolderIdList.end())).values();
+#else
 //        const QList<int> removedNoteSubFolderIdList =
```
Comment (pbek) [inline src/mainwindow.cpp:3310]: The "else" part is: Variable `removedNoteSubFolderIdList` will not be set at all? :laughing:

Comment (Waqar144) [reply src/mainwindow.cpp]: ah yes. Done :p

**Comment by pbek** [comment]
I hope you tested this. This is the most vital part of the application. :fearful:

**Comment by pbek** [comment]
Seems ok, so thank you. I'll keep my fingers crossed :crossed_fingers: :grin:

**Comment by Waqar144** [comment]
> I hope you tested this. This is the most vital part of the application. fearful

Dont worry. It will work. And if for some unexplainable reason it doesn't then the future of Qt is grim lol

**Pull request merged**